lib/Doctrine/DBAL/Schema/ForeignKeyConstraint.php

```diff
@@ -366,7 +366,7 @@ public function intersectsIndexColumns(Index $index)
     {
         foreach ($index->getUnquotedColumns() as $indexColumn) {
             foreach ($this->_localColumnNames as $localColumn) {
-                if (strtolower($indexColumn) === strtolower($localColumn->trimQuotes($localColumn->getName()))) {
```

Comment:
it seems this was not required `testIntersectsIndexEscapedColumns` should cover this